worlds/checksmate/Rules.py

```diff
@@ -0,0 +1,248 @@
+    #              lambda state, v=letter: has_enemy(state, "Capture Pawn " + str(v), player))
+    # names_to_capture = piece_names
+    # if not super_sized:
+    #     names_to_capture = piece_names_small
+    # for piece_name in names_to_capture:
+    #     add_rule(multiworld.get_location("Capture Piece " + piece_name, player),
+    #              lambda state, v=piece_name: has_enemy(state, "Capture Piece " + v, player))
+
+    # tactics
+    # add_rule(multiworld.get_location("Pin", player), lambda state: has_pin(state, player))
+    if opts.enable_tactics.value == opts.enable_tactics.option_all:
+        add_rule(multiworld.get_location("Fork, Sacrificial", player), lambda state: has_pin(state, player))
+        add_rule(multiworld.get_location("Fork, True", player), lambda state: has_pin(state, player))
+        add_rule(multiworld.get_location("Fork, Sacrificial Triple", player), lambda state: has_pin(state, player))
+        add_rule(multiworld.get_location("Fork, True Triple", player), lambda state: has_pin(state, player))
+        add_rule(multiworld.get_location("Fork, Sacrificial Royal", player), lambda state: has_pin(state, player))
+        add_rule(multiworld.get_location("Fork, True Royal", player), lambda state: has_pin(state, player))
+    add_rule(multiworld.get_location("Threaten Pawn", player), lambda state: count_enemy_pawns(state, player) > 0)
+    add_rule(multiworld.get_location("Threaten Minor", player), lambda state: count_enemy_pieces(state, player) > 3)
+    add_rule(multiworld.get_location("Threaten Minor", player), lambda state: has_pin(state, player))
+    add_rule(multiworld.get_location("Threaten Major", player), lambda state: count_enemy_pieces(state, player) > 5)
+    add_rule(multiworld.get_location("Threaten Major", player), lambda state: has_pin(state, player))
+    add_rule(multiworld.get_location("Threaten Queen", player), lambda state: count_enemy_pieces(state, player) > 6)
+    add_rule(multiworld.get_location("Threaten Queen", player), lambda state: has_pin(state, player))
+    add_rule(multiworld.get_location("Threaten King", player), lambda state: has_pin(state, player))
+    # special moves
+    total_queens = multiworld.worlds[player].items_used[player].get("Progressive Major To Queen", 0)
+    add_rule(multiworld.get_location("O-O Castle", player),
+             lambda state: state.count("Progressive Major Piece", player) >= 2 + total_queens)
+    add_rule(multiworld.get_location("O-O-O Castle", player),
+             lambda state: state.count("Progressive Major Piece", player) >= 2 + total_queens)
+    # add_rule(multiworld.get_location("French Move", player), lambda state: state.has_french_move(player))
+
+    # state cannot have super-size me for small checkmate
+    # todo: remove from the pool if non-ordered progressive
+    add_rule(multiworld.get_location("Checkmate Minima", player),
+             lambda state: not state.has("Super Size Me", player))
```

Comment:
This does not work. AP cannot work around `not state.has(...)`. In general, it can never work around gaining an item resulting in losing access to a location, the entire Fill algorithm is written around the assumption of increasing location accessibility per collected item (or staying the same ofc, >= 0 new locations per item)

Reply:
Oh, I can actually remove this now, the client supports visiting the old board. This shouldn't be an issue.

Thanks.